packages/expo-updates/src/Updates.types.ts

```diff
@@ -38,32 +38,56 @@ export type Manifest = ClassicManifest | typeof Constants.manifest2;
 /**
  * The result of checking for a new update.
  */
-export type UpdateCheckResult = {
-  /**
-   * `true` if an update is available, `false` if the app is already running the latest available
-   * update.
-   */
-  isAvailable: boolean;
-  /**
-   * If `isAvailable` is `true`, the manifest of the available update, and `undefined` otherwise.
-   */
-  manifest?: Manifest;
-};
+export type UpdateCheckResult =
+  | {
+      /**
+       * `true` if an update is available, `false` if the app is already running the latest available
+       * update.
+       */
+      isAvailable: true;
+      /**
+       * If `isAvailable` is `true`, the manifest of the available update, and `undefined` otherwise.
+       */
+      manifest: Manifest;
+    }
+  | {
+      /**
+       * `true` if an update is available, `false` if the app is already running the latest available
+       * update.
+       */
+      isAvailable: false;
+      /**
+       * If `isAvailable` is `true`, the manifest of the available update, and `undefined` otherwise.
+       */
+      manifest: undefined;
+    };
 
 /**
  * The result of fetching a new update.
  */
-export type UpdateFetchResult = {
-  /**
-   * `true` if the fetched bundle is new (that is, a different version than what's currently
-   * running), `false` otherwise.
-   */
-  isNew: boolean;
-  /**
-   * If `isNew` is `true`, the manifest of the newly downloaded update, and `undefined` otherwise.
-   */
-  manifest?: Manifest;
-};
+export type UpdateFetchResult =
+  | {
+      /**
+       * `true` if the fetched bundle is new (that is, a different version than what's currently
+       * running), `false` otherwise.
+       */
+      isNew: true;
+      /**
+       * If `isNew` is `true`, the manifest of the newly downloaded update, and `undefined` otherwise.
+       */
+      manifest: Manifest;
+    }
+  | {
+      /**
+       * `true` if the fetched bundle is new (that is, a different version than what's currently
+       * running), `false` otherwise.
+       */
+      isNew: false;
+      /**
+       * If `isNew` is `true`, the manifest of the newly downloaded update, and `undefined` otherwise.
+       */
+      manifest: undefined;
```

Comment:
```suggestion
      /**
       * Signifies that the fetched bundle is new (that is, a different version than what's currently
       * running).
       */
      isNew: true;
      /**
       * The manifest of the newly downloaded update.
       */
      manifest: Manifest;
    }
  | {
      /**
       * Signifies that the fetched bundle is the same as version which is currently running.
       */
      isNew: false;
      /**
       * No manifest, since there is no update.
       */
      manifest: undefined;
```

An example of updated descriptions.